vividus-plugin-mobitru/src/main/java/org/vividus/mobitru/MobitruClient.java

```diff
@@ -0,0 +1,103 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.mobitru;
+
+import java.io.IOException;
+import java.util.function.UnaryOperator;
+
+import org.apache.http.HttpStatus;
+import org.apache.http.client.methods.HttpUriRequest;
+import org.apache.http.entity.ContentType;
+import org.springframework.beans.factory.InitializingBean;
+import org.vividus.http.HttpMethod;
+import org.vividus.http.HttpRequestBuilder;
+import org.vividus.http.client.HttpResponse;
+import org.vividus.http.client.IHttpClient;
+
+public class MobitruClient implements InitializingBean
+{
+    private final IHttpClient httpClient;
+
+    private String apiUrl;
+    private String billingUnit;
+    private String apiBasePath;
+
+    public MobitruClient(IHttpClient httpClient)
+    {
+        this.httpClient = httpClient;
+    }
+
+    @Override
+    public void afterPropertiesSet()
+    {
+        this.apiBasePath = String.format("/billing/unit/%s/automation/api", billingUnit);
+    }
+
+    public byte[] takeDevice(String requestedDevice) throws IOException
+    {
+        return executePost("/device", requestedDevice, HttpStatus.SC_OK);
+    }
+
+    private byte[] executePost(String relativeUrl, String payload, int expectedResponseCode) throws IOException
+    {
+        return executeRequest(relativeUrl, HttpMethod.POST,
+                rb -> rb.withContent(payload, ContentType.APPLICATION_JSON), expectedResponseCode);
+    }
+
+    private byte[] executeRequest(String relativeUrl, HttpMethod httpMethod,
+        UnaryOperator<HttpRequestBuilder> configurator, int expectedResponseCode) throws IOException
+    {
+        HttpUriRequest request = configurator.apply(HttpRequestBuilder.create()
+                .withEndpoint(apiUrl)
+                .withHttpMethod(httpMethod)
+                .withRelativeUrl(apiBasePath + relativeUrl))
+                .build();
+        HttpResponse httpResponse = httpClient.execute(request);
+        httpResponse.verifyStatusCode(expectedResponseCode);
+        return httpResponse.getResponseBody();
+    }
+
+    public byte[] getApps() throws IOException
```

Comment:
move all public methods to the top, to avoid mixing internal and external class logic

Reply:
Doubtful statement from my POV :)